rxrabbit/src/main/java/com/meltwater/rxrabbit/impl/SingleChannelConsumer.java

```diff
@@ -150,15 +149,20 @@ private synchronized void startConsuming(Subscriber<? super Message> subscriber,
         );
         InternalConsumer cons;
         String threadNamePrefix = "consume-thread-" + consumerCount;
-        if (consumerRef.get()==null) {
+        if (consumerRef.get() == null) {
             cons = new InternalConsumer(channel, subscriber, closeTimeout, threadNamePrefix, metricsReporter, new AtomicLong(), new AtomicLong());
-        }else{
+        } else {
             cons = new InternalConsumer(consumerRef.get(), threadNamePrefix, channel, subscriber);
         }
-        channel.basicConsume(
-                consumerTag,
-                cons);
         consumerRef.set(cons);
```

Comment:
This reordering is important since when the InternalCustomer is closed it will try to close the channel. If we do not perform this operation before closing the InternalCustomer, we will get a leaked channel.